cdc/sink/ddlsink/mysql/mysql_ddl_sink.go

```diff
@@ -199,8 +199,12 @@ func (m *DDLSink) execDDL(pctx context.Context, ddl *model.DDLEvent) error {
 	}
 
 	if shouldSwitchDB {
-		_, err = tx.ExecContext(ctx, "USE "+quotes.QuoteName(ddl.TableInfo.TableName.Schema)+";")
+		stmt, err := tx.PrepareContext(ctx, "USE "+quotes.QuoteName(ddl.TableInfo.TableName.Schema)+";")
 		if err != nil {
+			log.Error("Failed to creates a prepared statement", zap.String("query", "USE "+quotes.QuoteName(ddl.TableInfo.TableName.Schema)+";"))
+			return err
+		}
+		if _, err = stmt.ExecContext(ctx); err != nil {
```

Comment:
1. it is obvious that this statement has no SQL injection opportunity
2. even if it is possible to inject through `QuoteName()`, I don't see how changing `Exec` to `Prepare` the entire statement unchanged and then `Exec` it is going to prevent anything.

`PrepareContext` can prevent SQL injection only when you replace e.g. `Exec("select x where y = " + val)` to `Prepare("select x where y = ?").Exec(val)`[^1]. Simply changing it to `Prepare("select x where y = " + val).Exec()` is meaningless.

[^1]: in Go you could even short-hand it as `Exec("select x where y = ?", val)`.